workflow-service/src/main/java/com/redhat/parodos/project/controller/ProjectController.java

```diff
@@ -104,4 +107,33 @@ public ResponseEntity<ProjectResponseDTO> getProjectById(@PathVariable UUID id)
 		return ResponseEntity.ok(projectResponseDTO);
 	}
 
+	@Operation(summary = "Add user roles to project")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Succeeded",
+					content = { @Content(mediaType = "application/json",
+							schema = @Schema(implementation = ProjectUserRoleResponseDTO.class)) }),
+			@ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content),
+			@ApiResponse(responseCode = "404", description = "Not found", content = @Content) })
+	@PostMapping("/{id}/userroles")
+	public ResponseEntity<ProjectUserRoleResponseDTO> updateUserRolesToProject(@PathVariable UUID id,
+			@RequestBody List<UserRoleRequestDTO> userRoleRequestDTOs) {
+		ProjectUserRoleResponseDTO projectResponseDTO = projectService.updateUserRolesToProject(id,
+				userRoleRequestDTOs);
+		return ResponseEntity.ok(projectResponseDTO);
+	}
+
+	@Operation(summary = "Remove users from project")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Succeeded",
+					content = { @Content(mediaType = "application/json",
+							schema = @Schema(implementation = ProjectUserRoleResponseDTO.class)) }),
+			@ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content),
+			@ApiResponse(responseCode = "404", description = "Not found", content = @Content) })
+	@DeleteMapping("/{id}/users")
+	public ResponseEntity<ProjectUserRoleResponseDTO> removeUsersFromProject(@PathVariable UUID id,
+			@RequestBody List<String> usernames) {
+		ProjectUserRoleResponseDTO projectUserRoleResponseDTO = projectService.removeUsersFromProject(id, usernames);
+		return ResponseEntity.ok(projectUserRoleResponseDTO);
+	}
+
```

Comment:
What about update and list?

Reply:
to list the users left in the project?  the response here will return all the updated data of this project where the given users are removed